### PR TITLE
Added Custom Policy definitions and documentations for B2C tenant, MDC Security Contacts and Kubernetes AAD.

### PR DESCRIPTION
**Pull request by saurabhgautam-msft**

### CODE REVIEW

File: Policies/KubernetesService/Azure_KubernetesService_AuthN_Enabled_AAD/Readme.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+## Azure_KubernetesService_AuthN_Enabled_AAD
+
+### DisplayName 
+[Remove Azure Active Directory B2C tenant(s) in a subscription](../../../Control%20coverage/Feature/KubernetesService.md#Azure_KubernetesService_AuthN_Enabled_AAD)
```
Comment (TarunKrShukla) [inline Policies/KubernetesService/Azure_KubernetesService_AuthN_Enabled_AAD/Readme.md:4]: The control file name is for AKS - AAD control and display name is different.

File: Policies/KubernetesService/Azure_KubernetesService_AuthN_Enabled_AAD/Security - Kubernetes Service - EnableAADAuthentication.json [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+            "not":{
+              "anyof":[
+                {
+                  "allof":[
+                    {
+                      "field": "Microsoft.ContainerService/managedClusters/aadProfile.managed",
+                      "exists": "true"
+                    },
+                    {
+                      "field": "Microsoft.ContainerService/managedClusters/aadProfile.managed",
```
Comment (TarunKrShukla) [inline Policies/KubernetesService/Azure_KubernetesService_AuthN_Enabled_AAD/Security - Kubernetes Service - EnableAADAuthentication.json:36]: Both conditions are exactly same is this intdended?

File: Policies/Subscription/Azure_Subscription_SI_Dont_Use_B2C_Tenant/Security - Subscription - DontUseB2CTenant.json [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        },
+        "allowedValues": [
+          "Audit",
+          "Disabled",
+          "Deny"
+        ],
+        "defaultValue": "Audit"
+      }
+    },
+    "policyRule": {
```
Comment (TarunKrShukla) [inline Policies/Subscription/Azure_Subscription_SI_Dont_Use_B2C_Tenant/Security - Subscription - DontUseB2CTenant.json:18]: As this is partial let's not push this policy.

File: Policies/KubernetesService/Azure_KubernetesService_AuthN_Enabled_AAD/Readme.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
+## Azure_KubernetesService_AuthN_Enabled_AAD
```
Comment (TarunKrShukla) [inline Policies/KubernetesService/Azure_KubernetesService_AuthN_Enabled_AAD/Readme.md:1]: We should also add this control and link to policy page in policy index/readme.md file as well.

File: Policies/Subscription/Azure_Subscription_SI_Dont_Use_B2C_Tenant/Readme.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
+## Azure_Subscription_SI_Dont_Use_B2C_Tenant
+
+### DisplayName 
+[Remove Azure Active Directory B2C tenant(s) in a subscription](../../../Control%20coverage/Feature/SubscriptionCore.md#Azure_Subscription_SI_Dont_Use_B2C_Tenant)
```
Comment (TarunKrShukla) [inline Policies/Subscription/Azure_Subscription_SI_Dont_Use_B2C_Tenant/Readme.md:4]: As policy to cover this  control is partial, let's not push this policy.

File: Policies/Subscription/Azure_Subscription_Config_MDC_Setup_SecurityContacts/Readme.md [hunk 6]
```diff
@@ -0,0 +1,24 @@
+## Azure_Subscription_Config_MDC_Setup_SecurityContacts
```
Comment (TarunKrShukla) [inline Policies/Subscription/Azure_Subscription_Config_MDC_Setup_SecurityContacts/Readme.md:1]: We should also add this control and link to policy page in policy index/readme.md file as well.

File: Policies/KubernetesService/Azure_KubernetesService_AuthN_Enabled_AAD/Security - Kubernetes Service - EnableAADAuthentication.json (hunk 2 above)
Comment (saurabhgautam-msft) [reply Policies/KubernetesService/Azure_KubernetesService_AuthN_Enabled_AAD/Security - Kubernetes Service - EnableAADAuthentication.json:36]: Not same,
one checks if the field exists and other check if it equals to true or not.

File: Policies/KubernetesService/Azure_KubernetesService_AuthN_Enabled_AAD/Readme.md (hunk 1 above)
Comment (saurabhgautam-msft) [reply Policies/KubernetesService/Azure_KubernetesService_AuthN_Enabled_AAD/Readme.md:4]: updated

File: Policies/Subscription/Azure_Subscription_SI_Dont_Use_B2C_Tenant/Security - Subscription - DontUseB2CTenant.json (hunk 3 above)
Comment (saurabhgautam-msft) [reply Policies/Subscription/Azure_Subscription_SI_Dont_Use_B2C_Tenant/Security - Subscription - DontUseB2CTenant.json:18]: reverting back

File: Policies/Subscription/Azure_Subscription_SI_Dont_Use_B2C_Tenant/Readme.md (hunk 5 above)
Comment (saurabhgautam-msft) [reply Policies/Subscription/Azure_Subscription_SI_Dont_Use_B2C_Tenant/Readme.md:4]: reverting back

File: Policies/KubernetesService/Azure_KubernetesService_AuthN_Enabled_AAD/Readme.md (hunk 4 above)
Comment (saurabhgautam-msft) [reply Policies/KubernetesService/Azure_KubernetesService_AuthN_Enabled_AAD/Readme.md:1]: Added

File: Policies/Subscription/Azure_Subscription_Config_MDC_Setup_SecurityContacts/Readme.md (hunk 6 above)
Comment (saurabhgautam-msft) [reply Policies/Subscription/Azure_Subscription_Config_MDC_Setup_SecurityContacts/Readme.md:1]: Added

File: Policies/Subscription/Azure_Subscription_Config_MDC_Setup_SecurityContacts/Security - Subscription - ConfigMDCSecurityContacts.json [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+        "allowedValues": [
+          "Audit",
+          "Disabled",
+          "Deny"
+        ],
+        "defaultValue": "Audit"
+      }
+    },
+    "policyRule": {
+      "if": {
```
Comment (TarunKrShukla) [inline Policies/Subscription/Azure_Subscription_Config_MDC_Setup_SecurityContacts/Security - Subscription - ConfigMDCSecurityContacts.json:19]: Which resource type will be flagged by this policyy

File: Policies/KubernetesService/Azure_KubernetesService_AuthN_Enabled_AAD/Security - Kubernetes Service - EnableAADAuthentication.json [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+    
+    "policyRule": {
+      "if": {
+        "allof":[
+          {
+            "field": "type",
+            "equals": "Microsoft.ContainerService/managedClusters"
+          },
+          {
+            "not":{
```
[the remark this answers is not in the document]
Comment (saurabhgautam-msft) [reply Policies/KubernetesService/Azure_KubernetesService_AuthN_Enabled_AAD/Security - Kubernetes Service - EnableAADAuthentication.json]: Updated

File: Policies/Subscription/Azure_Subscription_Config_MDC_Setup_SecurityContacts/Readme.md [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+Following policy can be used with 'Audit' effect to detect non-compliant resources.
+
+#### Policy Definition
+[Security - Subscription - ConfigMDCSecurityContacts](./Security%20-%20Subscription%20-%20ConfigMDCSecurityContacts)
+
+#### Parameter details
+
+|Param Name|Description|Default Value|Mandatory?
+|----|----|----|----|
+| Effect | The effect determines what happens when the policy rule is evaluated to match| Audit |No |
```
Comment (TarunKrShukla) [inline Policies/Subscription/Azure_Subscription_Config_MDC_Setup_SecurityContacts/Readme.md:20]: AuditIfNotExists

Comment (saurabhgautam-msft) [reply Policies/Subscription/Azure_Subscription_Config_MDC_Setup_SecurityContacts/Readme.md:20]: Updated. Forgot to update here.

File: Policies/Subscription/Azure_Subscription_Config_MDC_Setup_SecurityContacts/Security - Subscription - ConfigMDCSecurityContacts.json (hunk 7 above)
Comment (saurabhgautam-msft) [reply Policies/Subscription/Azure_Subscription_Config_MDC_Setup_SecurityContacts/Security - Subscription - ConfigMDCSecurityContacts.json:18]: Updated, we are targeting subscription here.

**Review by TarunKrShukla — APPROVED** [review]
Looks good.

**Review by jaiswalricha — APPROVED** [review]

File: Policies/KubernetesService/Azure_KubernetesService_AuthN_Enabled_AAD/Security - Kubernetes Service - EnableAADAuthentication.json [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+          },
+          {
+            "anyof":[
+              {
+                "field": "Microsoft.ContainerService/managedClusters/aadProfile.clientAppID",
+                "exists": "false"
+              },
+              {
+                "field": "Microsoft.ContainerService/managedClusters/aadProfile.clientAppID",
+                "Equals": ""
```
[the remark this answers is not in the document]
Comment (saurabhgautam-msft) [reply Policies/KubernetesService/Azure_KubernetesService_AuthN_Enabled_AAD/Security - Kubernetes Service - EnableAADAuthentication.json]: Addressed.

**Pull request merged**